examples/apps/table-detail-tea/src/main.rs

```diff
@@ -0,0 +1,211 @@
+/// release.
+///
+/// [`latest`]: https://github.com/ratatui/ratatui/tree/latest
+use std::io::stdout;
+use std::time::Duration;
+use std::{fmt, panic};
+
+use crossterm::event::{
+    self, DisableMouseCapture, EnableMouseCapture, Event, KeyCode, MouseEventKind,
+};
+use crossterm::ExecutableCommand;
+use fakeit::{address, contact, name};
+use ratatui::layout::{Constraint, Layout, Rect};
+use ratatui::style::{Modifier, Style};
+use ratatui::widgets::{Block, Padding, Paragraph, Row, Table, TableState};
+use ratatui::{DefaultTerminal, Frame};
+
+#[derive(Debug)]
+struct AppModel {
+    people: Vec<Person>,
+    table_state: TableState,
+}
+
+#[derive(Debug)]
+struct Person {
+    name: String,
+    address: Address,
+    email: String,
+}
+
+#[derive(Debug)]
+struct Address {
+    street: String,
+    city: String,
+    state: String,
+    zip: String,
+}
+
+#[derive(PartialEq, Copy, Clone)]
+enum Message {
+    SelectNext,
+    SelectPrevious,
+    Quit,
+}
+
+fn main() -> color_eyre::Result<()> {
+    let terminal = ratatui::init();
+    install_panic_hook();
+    stdout().execute(EnableMouseCapture)?;
+
+    let model = AppModel::default();
+    let result = model.run(terminal);
+
+    stdout().execute(DisableMouseCapture)?;
+    ratatui::restore();
+    result
+}
+
+pub fn install_panic_hook() {
+    let original_hook = panic::take_hook();
+    panic::set_hook(Box::new(move |panic_info| {
+        stdout().execute(DisableMouseCapture).ok();
+        original_hook(panic_info);
+    }));
+}
+
+impl Default for AppModel {
+    fn default() -> Self {
+        Self {
+            people: (0..50).map(|_| Person::fake()).collect(),
+            table_state: TableState::default().with_selected(0),
+        }
+    }
+}
+
+impl AppModel {
+    fn run(mut self, mut terminal: DefaultTerminal) -> color_eyre::Result<()> {
+        loop {
+            terminal.draw(|frame| self.view(frame))?;
+
+            let mut maybe_message = handle_event()?;
+            while let Some(message) = maybe_message {
+                if message == Message::Quit {
+                    return Ok(());
+                }
+                maybe_message = self.update(message);
+            }
+        }
+    }
+
+    fn update(&mut self, message: Message) -> Option<Message> {
+        match message {
+            Message::Quit => {}
+            Message::SelectNext => self.table_state.select_next(),
+            Message::SelectPrevious => self.table_state.select_previous(),
+        };
+        None
+    }
+
+    /// Render the current view
+    fn view(&mut self, frame: &mut Frame) {
+        let [top, bottom] = Layout::vertical([Constraint::Fill(1); 2]).areas(frame.area());
+        self.render_table(frame, top);
+        self.render_detail(frame, bottom);
+    }
+
+    fn render_table(&mut self, frame: &mut Frame, area: Rect) {
+        let header = Row::new(vec!["Name", "Address", "Email"])
+            .style(Modifier::BOLD)
+            .height(1);
```

Comment:
nit: What about moving the row header to the Person impl?
(is height 1 the default btw?)

E.g.:
```suggestion
        let header = Person::header_row().bold();
```

or 
```suggestion
        let header = Row::from(Person::headers()).bold();
```

Not sure which is nicer - just a thought (especially since there's a conversion from Person to Row already below)